clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -579,11 +629,34 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
             ListOffsetResponse.PartitionData partitionData = listOffsetResponse.responseData().get(topicPartition);
             Errors error = Errors.forCode(partitionData.errorCode);
             if (error == Errors.NONE) {
-                OffsetAndTimestamp offsetAndTimestamp = null;
-                if (partitionData.offset != ListOffsetResponse.UNKNOWN_OFFSET)
-                    offsetAndTimestamp = new OffsetAndTimestamp(partitionData.offset, partitionData.timestamp);
-                log.debug("Fetched {} for partition {}", offsetAndTimestamp, topicPartition);
-                timestampOffsetMap.put(topicPartition, offsetAndTimestamp);
+                if (partitionData.offsets != null) {
```

Comment:
Yeah, it would be good to have a clear pattern here.  Let's do it in a follow-on.